ui/packages/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js

```diff
@@ -23,9 +23,90 @@ const wreqr = require('../../exports/wreqr.js')
 const Router = require('../router/router.js')
 const announcement = require('../announcement')
 const Common = require('../../js/Common.js')
+import styled from '../../react-component/styles/styled-components'
+
+const FormTitle = styled.h3`
+  padding-bottom: ${props => props.theme.minimumSpacing};
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: inline-block;
+  width: 100%;
+`
+
+const FormContents = styled.span`
+  display: block;
+`
+
+const NewFormCircle = styled.div`
+  font-size: calc(3 * ${props => props.theme.largeFontSize});
+  padding-top: ${props => props.theme.minimumSpacing};
+`
+
+const DefaultIcon = styled.div`
+  position: absolute;
+  right: 0;
+  top: 0;
+`
+
+const CustomSearchForm = props => {
+  const { title, createdOn, createdBy, isDefault } = props
+  return (
+    <div style={{ position: 'relative', height: '100%' }}>
+      <FormTitle data-help={title}>{title}</FormTitle>
+      {isDefault ? <DefaultIcon className="fa fa-star" /> : null}
+      <FormContents>{createdOn}</FormContents>
+      <FormContents>
+        <span className="fa fa-cloud" />
+        {' ' + createdBy}
+      </FormContents>
+      <span
+        className="choice-actions is-button"
+        title="Shows a list of actions to take on the search forms"
+        data-help="Shows a list of actions to take on the search forms."
+      />
+    </div>
+  )
+}
 
 module.exports = Marionette.LayoutView.extend({
-  template: template,
+  template(props) {
+    const isDefault = user.getQuerySettings().isTemplate(this.model)
+    if (props.type === 'custom') {
+      return (
+        <CustomSearchForm
+          {...props}
+          isDefault={isDefault}
+          onClick={this.changeView}
+        />
+      )
+    } else if (props.type === 'new-result') {
+      return (
+        <div>
+          <NewFormCircle className="fa fa-plus-circle" />
```

Comment:
I'm a little confused by some of this code, I thought this was for search forms not result forms.  Is this component reused by both?

Reply:
Yes, they are currently coupled together here. I would love to decouple them but I think that would be out of scope for this ticket, which has already suffered from significant scope creep.